calib3d_test.go

```diff
@@ -383,6 +383,85 @@ func TestFindAndDrawChessboard(t *testing.T) {
 	}
 }
 
+func TestCalibrateCamera(t *testing.T) {
+	img := IMRead("images/chessboard_4x6_distort.png", IMReadGrayScale)
+	if img.Empty() {
+		t.Error("Invalid read of chessboard image")
+		return
+	}
+	defer img.Close()
+
+	corners := NewMat()
+	defer corners.Close()
+
+	size := image.Pt(4, 6)
+	found := FindChessboardCorners(img, size, &corners, 0)
+	if !found {
+		t.Error("chessboard pattern not found")
+		return
+	}
+	if corners.Empty() {
+		t.Error("chessboard pattern not found")
+		return
+	}
+
+	imagePoints := NewPoint2fVectorFromMat(corners)
+	defer imagePoints.Close()
+
+	objectPoints := NewPoint3fVector()
+	defer objectPoints.Close()
+
+	for j := 0; j < size.Y; j++ {
+		for i := 0; i < size.X; i++ {
+			objectPoints.Append(Point3f{
+				X: float32(100 * i),
+				Y: float32(100 * j),
+				Z: 0,
+			})
+		}
+	}
+
+	cameraMatrix := NewMat()
+	defer cameraMatrix.Close()
+	distCoeffs := NewMat()
+	defer distCoeffs.Close()
+	rvecs := NewMat()
+	defer rvecs.Close()
+	tvecs := NewMat()
+	defer tvecs.Close()
+
+	objectPointsVector := NewPoints3fVector()
+	objectPointsVector.Append(objectPoints)
+	defer objectPointsVector.Close()
+
+	imagePointsVector := NewPoints2fVector()
+	imagePointsVector.Append(imagePoints)
+	defer imagePointsVector.Close()
+
+	CalibrateCamera(
+		objectPointsVector, imagePointsVector, image.Pt(img.Cols(), img.Rows()),
+		&cameraMatrix, &distCoeffs, &rvecs, &tvecs, 0,
+	)
+
+	dest := NewMat()
+	defer dest.Close()
+	Undistort(img, &dest, cameraMatrix, distCoeffs, cameraMatrix)
+
+	target := IMRead("images/chessboard_4x6_distort_correct.png", IMReadGrayScale)
+	defer target.Close()
+
+	xor := NewMat()
+	defer xor.Close()
+
+	// The method for compare is ugly : different pix number < 0.5%
+	BitwiseXor(dest, target, &xor)
```

Comment:
Hello @deadprogram, please help me to see whether the code is okay for comparing difference between two images.

Reply:
The "best" way would be to use something like MSE or SSIM.

That would require yet another OpenCV modules that is not yet supported in GoCV :smiley_cat: 
https://github.com/opencv/opencv_contrib/tree/4a36e77dba0f8f207b5bd803deefbca2d497d190/modules/quality

In other words, any test at all is probably fine for now.